File: lib/screens/landing_screen.dart
import 'package:flutter/material.dart';
import 'package:image_picker/image_picker.dart';
import 'package:provider/provider.dart';
import '../services/image_picker_service.dart';
import '../services/ocr_service.dart';
import '../models/transaction.dart';
import '../providers/transaction_provider.dart';
import 'dashboard_screen.dart';

class LandingScreen extends StatefulWidget {
  const LandingScreen({super.key});

  @override
  State<LandingScreen> createState() => _LandingScreenState();
}

class _LandingScreenState extends State<LandingScreen> {
  final ImagePickerService _imagePickerService = ImagePickerService();
  final OCRService _ocrService = OCRService();
  bool _isProcessing = false;

  Future<void> _handleAddReceipt() async {
    setState(() => _isProcessing = true);
    
    try {
      final image = await _imagePickerService.pickImage(ImageSource.gallery);
      
      if (image != null && mounted) {
        // Process OCR
        final extractedText = await _ocrService.extractText(image.path);
        
        if (mounted) {
          // Parse receipt data
          final receiptData = await _ocrService.parseReceiptData(extractedText);
          
          // Show extracted data in a dialog
          showDialog(
            context: context,
            builder: (context) => AlertDialog(
              title: const Text('Receipt Scanned'),
              content: SingleChildScrollView(
                child: Column(
                  crossAxisAlignment: CrossAxisAlignment.start,
                  mainAxisSize: MainAxisSize.min,
                  children: [
                    Text('Merchant: ${receiptData['merchant']}', style: const TextStyle(fontWeight: FontWeight.bold)),
                    const SizedBox(height: 8),
                    Text('Amount: \$${receiptData['amount'].toStringAsFixed(2)}', style: const TextStyle(fontWeight: FontWeight.bold)),
                    const SizedBox(height: 16),
                    const Text('Raw Text:', style: TextStyle(fontWeight: FontWeight.bold)),
                    const SizedBox(height: 4),
                    Text(extractedText.isEmpty ? 'No text found' : extractedText, style: const TextStyle(fontSize: 12)),
                  ],
                ),
              ),
              actions: [
                TextButton(
                  onPressed: () => Navigator.pop(context),
                  child: const Text('Cancel'),
                ),
                TextButton(
                  onPressed: () {
                    Navigator.pop(context);
                    // Add transaction to provider
                    final now = DateTime.now();
                    final formattedDate = "${now.month.toString().padLeft(2, '0')}/${now.day.toString().padLeft(2, '0')}/${now.year}";

                    final transaction = Transaction(
                      id: now.millisecondsSinceEpoch.toString(),
                      name: receiptData['merchant'],
                      category: 'Shopping', // Default category
                      date: formattedDate,
                      amount: -receiptData['amount'], // Negative for expense
                      type: 'Expense',
                    );
                    
                    Provider.of<TransactionProvider>(context, listen: false).addTransaction(transaction);
                    
                    ScaffoldMessenger.of(context).showSnackBar(
                      const SnackBar(content: Text('Transaction added successfully!')),
                    );
                  },
                  child: const Text('Add Transaction'),
                ),
              ],
            ),
          );
        }
      }
    } catch (e) {
      if (mounted) {
        ScaffoldMessenger.of(context).showSnackBar(
          SnackBar(content: Text('Error: ${e.toString()}')),
        );
      }
    } finally {
      if (mounted) {
        setState(() => _isProcessing = false);
      }
    }
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: const Text('Dashboard', style: TextStyle(fontWeight: FontWeight.w400)),
        centerTitle: true,
        elevation: 0,
        backgroundColor: Colors.white,
        foregroundColor: Colors.black,
      ),
      body: SingleChildScrollView(
        padding: const EdgeInsets.all(24),
        child: Column(
          crossAxisAlignment: CrossAxisAlignment.start,
          children: [
            // spend.AI branding
            const Text(
              'spend.AI',
              style: TextStyle(
                fontSize: 36,
                fontWeight: FontWeight.bold,
                letterSpacing: -1,
              ),
            ),
            const SizedBox(height: 40),

            // Connected Cards Section
            const Text(
              'Connected Cards',
              style: TextStyle(
                fontSize: 18,
                fontWeight: FontWeight.w600,
              ),
            ),
            const SizedBox(height: 12),
            Text(
              'No cards yet. Add one to get started.',
              style: TextStyle(
                fontSize: 14,
                color: Colors.grey[600],
              ),
            ),
            const SizedBox(height: 16),
            
            // Add New Card Button
            _DashedButton(
              onPressed: () {
                ScaffoldMessenger.of(context).showSnackBar(
                  const SnackBar(content: Text('Card integration coming soon!')),
                );
              },
              icon: Icons.add,
              label: 'Add New Card',
            ),
            const SizedBox(height: 40),

            // Spending by Date Section
            const Text(
              'Spending by Date',
              style: TextStyle(
                fontSize: 18,
                fontWeight: FontWeight.w600,
              ),
            ),
            const SizedBox(height: 12),
            Text(
              'No receipts yet. Add one to see your spending.',
              style: TextStyle(
                fontSize: 14,
                color: Colors.grey[600],
              ),
            ),
            const SizedBox(height: 16),
            
            // Add Receipt Button
            _DashedButton(
              onPressed: _isProcessing ? null : _handleAddReceipt,
              icon: Icons.camera_alt,
              label: _isProcessing ? 'Processing...' : 'Add Receipt from Photos',
            ),
            const SizedBox(height: 40),

            // Navigate to Dashboard Button
            SizedBox(
              width: double.infinity,
              child: ElevatedButton(
                onPressed: () {
                  Navigator.of(context).push(
                    MaterialPageRoute(builder: (context) => const DashboardScreen()),
                  );
                },
                style: ElevatedButton.styleFrom(
                  backgroundColor: Colors.black,
                  foregroundColor: Colors.white,
                  padding: const EdgeInsets.symmetric(vertical: 16),
                  shape: RoundedRectangleBorder(
                    borderRadius: BorderRadius.circular(12),
                  ),
                ),
                child: const Text('View Full Dashboard'),
              ),
            ),
          ],
        ),
      ),
    );
  }
}

class _DashedButton extends StatelessWidget {
  final VoidCallback? onPressed;
  final IconData icon;
  final String label;

  const _DashedButton({
    required this.onPressed,
    required this.icon,
    required this.label,
  });

  @override
  Widget build(BuildContext context) {
    return InkWell(
      onTap: onPressed,
      borderRadius: BorderRadius.circular(12),
      child: Container(
        width: double.infinity,
        padding: const EdgeInsets.symmetric(vertical: 20, horizontal: 16),
        decoration: BoxDecoration(
          border: Border.all(
            color: const Color(0xFF2196F3),
            width: 2,
            strokeAlign: BorderSide.strokeAlignInside,
          ),
          borderRadius: BorderRadius.circular(12),
          color: Colors.white,
        ),
        child: Row(
          mainAxisAlignment: MainAxisAlignment.center,
          children: [
            Icon(icon, color: const Color(0xFF2196F3), size: 20),
            const SizedBox(width: 8),
            Text(
              label,
              style: const TextStyle(
                color: Color(0xFF2196F3),
                fontSize: 16,
                fontWeight: FontWeight.w500,
              ),
            ),
          ],
        ),
      ),
    );
  }
}

File: android/app/src/main/kotlin/com/example/spendai/ui/LandingScreen.kt
package com.example.spendai.ui

import android.net.Uri
import android.widget.Toast
import androidx.activity.compose.rememberLauncherForActivityResult
import androidx.activity.result.PickVisualMediaRequest
import androidx.activity.result.contract.ActivityResultContracts
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Add
import androidx.compose.material.icons.filled.CameraAlt
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.lifecycle.viewmodel.compose.viewModel
import com.example.spendai.models.Transaction
import com.example.spendai.services.OCRService
import com.example.spendai.viewmodels.TransactionViewModel
import kotlinx.coroutines.launch

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun LandingScreen(
    onNavigateToDashboard: () -> Unit,
    viewModel: TransactionViewModel = viewModel()
) {
    val context = LocalContext.current
    val scope = rememberCoroutineScope()
    val ocrService = remember { OCRService(context) }
    var isProcessing by remember { mutableStateOf(false) }

    val photoPickerLauncher = rememberLauncherForActivityResult(
        contract = ActivityResultContracts.PickVisualMedia(),
        onResult = { uri ->
            if (uri != null) {
                isProcessing = true
                scope.launch {
                    try {
                        val text = ocrService.extractText(uri)
                        val data = ocrService.parseReceiptData(text)
                        
                        val merchant = data["merchant"] as String
                        val amount = data["amount"] as Double
                        val date = data["date"] as String

                        val newTransaction = Transaction(
                            name = merchant,
                            category = "Shopping",
                            date = date,
                            amount = -amount,
                            type = "Expense"
                        )

                        if (viewModel.isDuplicate(newTransaction)) {
                            Toast.makeText(context, "Duplicate transaction detected!", Toast.LENGTH_LONG).show()
                        } else {
                            viewModel.addTransaction(newTransaction)
                            Toast.makeText(context, "Transaction added successfully!", Toast.LENGTH_SHORT).show()
                        }
                    } catch (e: Exception) {
                        Toast.makeText(context, "Error: ${e.message}", Toast.LENGTH_SHORT).show()
                    } finally {
                        isProcessing = false
                    }
                }
            }
        }
    )

    var showAssistant by remember { mutableStateOf(false) }

    if (showAssistant) {
        ModalBottomSheet(
            onDismissRequest = { showAssistant = false },
            containerColor = Color.White
        ) {
            AssistantPanel(onClose = { showAssistant = false })
        }
    }

    Scaffold(
        floatingActionButton = {
            AssistantFab(onPressed = { showAssistant = true })
        },
        topBar = {
            CenterAlignedTopAppBar(
                title = { Text("Dashboard", fontWeight = FontWeight.Normal) },
                colors = TopAppBarDefaults.centerAlignedTopAppBarColors(
                    containerColor = Color.White,
                    titleContentColor = Color.Black
                )
            )
        }
    ) { paddingValues ->
        Column(
            modifier = Modifier
                .padding(paddingValues)
                .padding(24.dp)
                .verticalScroll(rememberScrollState())
        ) {
            Text(
                text = "spendAI",
                fontSize = 36.sp,
                fontWeight = FontWeight.Bold,
                letterSpacing = (-1).sp
            )
            Spacer(modifier = Modifier.height(40.dp))

            Text(
                text = "Connected Cards",
                fontSize = 18.sp,
                fontWeight = FontWeight.SemiBold
            )
            Spacer(modifier = Modifier.height(12.dp))
            Text(
                text = "No cards yet. Add one to get started.",
                fontSize = 14.sp,
                color = Color.Gray
            )
            Spacer(modifier = Modifier.height(16.dp))

            DashedButton(
                onClick = { Toast.makeText(context, "Card integration coming soon!", Toast.LENGTH_SHORT).show() },
                icon = Icons.Default.Add,
                label = "Add New Card"
            )
            Spacer(modifier = Modifier.height(40.dp))

            Text(
                text = "Spending by Date",
                fontSize = 18.sp,
                fontWeight = FontWeight.SemiBold
            )
            Spacer(modifier = Modifier.height(12.dp))
            Text(
                text = "No receipts yet. Add one to see your spending.",
                fontSize = 14.sp,
                color = Color.Gray
            )
            Spacer(modifier = Modifier.height(16.dp))

            DashedButton(
                onClick = {
                    if (!isProcessing) {
                        photoPickerLauncher.launch(
                            PickVisualMediaRequest(ActivityResultContracts.PickVisualMedia.ImageOnly)
                        )
                    }
                },
                icon = Icons.Default.CameraAlt,
                label = if (isProcessing) "Processing..." else "Add Receipt from Photos"
            )
            Spacer(modifier = Modifier.height(40.dp))

            Button(
                onClick = onNavigateToDashboard,
                modifier = Modifier.fillMaxWidth(),
                shape = RoundedCornerShape(12.dp),
                colors = ButtonDefaults.buttonColors(
                    containerColor = Color.Black,
                    contentColor = Color.White
                ),
                contentPadding = PaddingValues(vertical = 16.dp)
            ) {
                Text("View Full Dashboard")
            }
        }
    }
}

@Composable
fun DashedButton(
    onClick: () -> Unit,
    icon: ImageVector,
    label: String
) {
    // Note: Jetpack Compose doesn't have a built-in dashed border modifier easily accessible without custom drawing.
    // For simplicity, using a solid border with a specific color to match the design intent.
    Box(
        modifier = Modifier
            .fillMaxWidth()
            .border(
                width = 2.dp,
                color = Color(0xFF2196F3),
                shape = RoundedCornerShape(12.dp)
            )
            .clickable(onClick = onClick)
            .padding(vertical = 20.dp, horizontal = 16.dp),
        contentAlignment = Alignment.Center
    ) {
        Row(verticalAlignment = Alignment.CenterVertically) {
            Icon(
                imageVector = icon,
                contentDescription = null,
                tint = Color(0xFF2196F3),
                modifier = Modifier.size(20.dp)
            )
            Spacer(modifier = Modifier.width(8.dp))
            Text(
                text = label,
                color = Color(0xFF2196F3),
                fontSize = 16.sp,
                fontWeight = FontWeight.Medium
            )
        }
    }
}
